redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -452,6 +452,107 @@ inline bool setSeverity(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
     return true;
 }
 
+void assembleEventProperties(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                             const dbus::utility::DBusPropertiesMap& properties,
+                             nlohmann::json& condition,
+                             const std::string& hwStatusEventObj)
+{
+    using AssociationsValType =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+    const AssociationsValType* associations = nullptr;
+    const uint64_t* timestamp = nullptr;
+    const std::string* msgPropVal = nullptr;
+    const std::string* severity = nullptr;
+
+    const bool success = sdbusplus::unpackPropertiesNoThrow(
+        dbus_utils::UnpackErrorPrinter(), properties, "Associations",
```

Comment:
> dbus_utils::UnpackErrorPrinter()

In the old code, we are adding a trace with `BMCWEB_LOG_ERROR` if the property value is not found (nullptr) with the object path that we are trying to look but in the new code, are we adding a trace with `BMCWEB_LOG_DEBUG` and with the object path?

If yes, I would like to print in the ERROR log level (and object path) since we won't enable debug log level in the field (and also I think bmcweb debug log are huge so I think we disabled it, please confirm it).

Reply:
Just add the BMCWEB_LOG_ERROR on line 471 ?

Reply:
Would it be better if we update the dbus_utils::UnpackErrorPrinter() to log BMCWEB_LOG_ERROR. That way it will work everywhere and the code would be generic. @RameshIyyar Could you please share your preference?

Reply:
>Would it be better if we update the dbus_utils::UnpackErrorPrinter() to log BMCWEB_LOG_ERROR. That way it will work everywhere and the code would be generic.

I would prefer that only...

but, I am worried about the hw-isolation now, and even adding a trace in the "if (!success) {...}" won't solve the issue that is, how we will know which attribute failed to read?

Reply:
> if (!success) {...} won't solve the issue, right

It should though. How if we add  BMCWEB_LOG_ERROR to line 471 is this different than what we had before?


Before we had some nullptr checks, which we can add. sdbusplus::unpackPropertiesNoThrow is the new cleaner way to do this and I don't see how it has any less logging 

                        if (associations == nullptr)
                        {
                            BMCWEB_LOG_ERROR
                                << "Failed to get the Associations "
                                << "from object: " << hwStatusEventObj;
                            messages::internalError(aResp->res);
                            return;
                        }

Reply:
> sdbusplus::unpackPropertiesNoThrow is the new cleaner way to do this

Agree, but`UnpackErrorPrinter` was added in the debug log trace which is my concern. @gtmills cannot we use the error log trace in the `dbus_utils::UnpackErrorPrinter` function? what is the problem? any issue with it?

If we used the ERROR log trace in the `dbus_utils::UnpackErrorPrinter` then we are good since [it](https://github.com/ibm-openbmc/bmcweb/blob/1050/redfish-core/include/utils/dbus_utils.hpp#L18) prints the failed property if we are hitting failure while [reading a property](https://github.com/openbmc/sdbusplus/blob/af9556707029ca30e4c2749502f965fef28d5388/include/sdbusplus/unpack_properties.hpp#L31) value from the `properties` container and we could add the object path inside `if (!success) {...}` so no extra changes required else we need to follow the old code for each property (like you posted above).

If we are adding a check for each property and trace with an error log level then, I am fine with it.

(Just a suggestion, if we are adding a check like this for each property then we could avoid the `unpackPropertiesNoThrow` call in this function since it is not required, we could use the existing code `for` loop with `properties`).

Reply:
I added another commit to address this. 
With this Unpack will point to the error that it had issues reading. And internal error will log the source file name and line number. So with this generic solution we should be able to figure out most of issue!!